### ISSUE TITLE
Lot's of CPU usage while idle

### ISSUE THREAD

**Issue by Juburin** [MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-#920]
**OS:** OS X 10.11.4

**Issue Descriptions:** The player utilizes a lot of cpu even when idle. I don't have this problem with other electron apps and even chrome has a better cpu usage when i have https://play.google.com/music/listen open.

<img width="785" alt="bildschirmfoto 2016-04-08 um 14 30 19" src="https://cloud.githubusercontent.com/assets/767690/14383775/824e04d6-fd97-11e5-872f-471db87b2280.png">
<img width="782" alt="bildschirmfoto 2016-04-08 um 14 35 50" src="https://cloud.githubusercontent.com/assets/767690/14383777/82565e2e-fd97-11e5-8a6a-71e207e1f519.png">
<img width="772" alt="bildschirmfoto 2016-04-08 um 14 36 22" src="https://cloud.githubusercontent.com/assets/767690/14383776/824ee0d6-fd97-11e5-99ba-713314fbcf53.png">

**Steps to Reproduce:** Run on OS X, check CPU usage

**Comment by ahatzz11** [comment]
I am also having this issue. I'm not sure why there are so many processes running for this one app, but it is consistently using a large portion of my CPU. 

![](http://puu.sh/oaQ9m/c4ff4f4b92.png)

![](http://puu.sh/oaQfG/0eaa540f52.png)

**Steps to Reproduce:** Run on OS X, check CPU usage

**Comment by Double44** [comment]
I noticed this too, particularly when the application is in the foreground.

**Comment by MarshallOfSound** [comment]
Do you guys have voice controls turned on?
What are the specs of your mac?

**Comment by Double44** [comment]
I forgot to mention I'm on Windows 10..
![capture](https://cloud.githubusercontent.com/assets/6753152/14407887/be1709da-fe97-11e5-9696-f09ede5283d9.PNG)

**Comment by MarshallOfSound** [comment]
I use windows 10 as my daily driver and the player uses on average `0.2%` CPU when inactive and in focus and `1.1%` when active and in focus.

**Comment by Double44** [comment]
The top process can reach up to 20%, but the second Task Manager and Snipping Tool becomes foreground, the player calms down.

**Comment by ahatzz11** [comment]
@MarshallOfSound I do not have voice control turned on. 

Macbook Pro 15" - 2.3 GHz Intel Core i7, 16GB 1600 MHz DDR3, Intel Iris Pro 1536 MB.

**Comment by Juburin** [comment]
@MarshallOfSound voice control is turned off.

MacBook Pro (Retina, 15", Late 2013), 2,3 GHz Intel Core i7, 16 GB 1600 MHz DDR3, NVIDIA GeForce GT 750M 2048 MB

**Comment by tuolumne** [comment]
I have similar issues on  Fedora Linux  - CPU usage is out the roof. This is while playing music however. 
![googlemusicdesktopapp-cpu-usage-2016-04-14 11 27 28](https://cloud.githubusercontent.com/assets/1174706/14539180/2aebad50-0234-11e6-9cd8-a027cbd1175a.png)

**Comment by MarshallOfSound** [comment]
@tuolumne Can you compare that somewhat insane CPU usage to listening to Google Play Music in Chrome with the Flash plugin disabled (Go to chrome://plugins and disable flash).  It's possibly that Chrome/Chromiums implementation of MP3 decoding is quite CPU intensive.

Also @Juburin @ahatzz11 can you try as well, ensure you disable flash player to force MP3 decoding

**Comment by tuolumne** [comment]
Same CPU usage when I use the browser. I believe this is actually not related to the sound at all but the graphics. As soon as I expand the album art the CPU goes out the roof.

**Comment by MarshallOfSound** [comment]
@tuolumne Are you running with the `--disable-gpu` flag?  That might explain the CPU offloading.  

Though in a sense it's good to hear chrome has the same issue, it means it's something either internally in chromium or on Google Play Music's side of things that is using up the CPU.

**Comment by justmyfault** [comment]
I am on Ubuntu 16.04 beta and I noticed the same problem on both Google Play Music Desktop and regular Google Chrome tabs... However the problem of CPU usage does not appear using Firefox!
This is extremely disappointing, I can't use Google Play Music when I am on battery....
Some people on the google product forums reported that deleting the entire library and reuploading it solves the problem... but as I have 35k files, I'd rather not try that...

**Comment by MarshallOfSound** [comment]
This sounds more and more like a not very efficient MP3 decoder inside Chromium.  I'm tempted to close this as a "Can't fix" but I'm going to do some more research first.

**Comment by justmyfault** [comment]
I do agree on the fact that it's not your app's fault, however I tried with chrome with flash and not html5, and I tried also enabling and disabling GPU hardware acceleration but the problem is still there...

**Comment by jostrander** [comment]
I think there are two separate issues going on in this thread.

@tuolumne mentions:

**Comment by MarshallOfSound** [comment]
@justmyfault Are you sure you were using flash.  GPM ignores your HTML5 audio setting most of the time, the only way to be sure is to check for audio elements on your webpage.  Google is strange like that

**Comment by MaxRink** [comment]
I'm also affected by that. Based upon my quick research it related to executed animations.

**Comment by animusyosho** [comment]
I seem to be having high CPU usage, and there are 3-5 instances of the application in my process tab (on windows 7 T44 laptop). As an added bonus, after several minutes of this the sound distorts and then the audio device remains hung until I either restart my computer or bounce the Windows Audio service. So far in testing this issue only occurs after several minutes of having this application open, and has only been happening as of today. I recall when I started the app today it updated to the new version, so I suspect a recent change, though I am also using new headphones which use an audio jack, so internal sound card, where before I was using a USB headset. I seem to have no problem when using the google chrome app, only  MarshallOfSound's desktop app. 
Thank you for the great app and for being attentive to issues!!

**Comment by Double44** [comment]
Yes playing music through the browser is nowhere near as intensive on the cpu as the app is. I haven't been able to pinpoint the exact cause, but my laptop fan has a tendency to act up every time the GUI is in focus, but it does manage to calm down over time, dunno why. The cpu usage is a problem for me because I can't hear my music over the fan. I was hoping the recent update might fix this, but its still happening.

**Comment by MarshallOfSound** [comment]
@Double44 It is possible your "browser" is using flash player so the CPU usage is off loaded to another process, or simply isn't as intensive as the native MP3 decoder.

**Comment by rfuhrer** [comment]
I'm having the same issue as people above.  I hit 25% (100% usage on a single core) CPU in an i5 3.4ghz when I first open the application.  Even before playing a song.  I hit this number again while scrolling on any screen.  If i have the application in the foreground it jumps around between 10-20% usage across the multiple processes it has running.  If I minimize it the usage drops to 0.5-2%.  Definitely seems like it's a rendering/animation issue.  I have all the settings at default (not forcing it to not use the GPU or anything like that).  Changing the window border, theme colours, light/dark mode doesn't affect the CPU usage.

I just did some more digging and I think this might be a google music issue.  While using the web browser does use way less CPU when playing, I think that's just an illusion from you usually switching tabs which kills the animations being processed.  Even dragging my mouse up and down on tracks in a playlist rapidly can make the CPU jump to 50%.  They just rolled out the new Play Music icon and the podcast section, I'm guessing this also came with some changes to the framework they're using for "material" design and animations.

TL;DR: This seems to be an issue from google, not this application.  Minimizing the desktop player brings CPU levels back down to the 0-2% you expect.

**Comment by MarshallOfSound** [comment]
After quite a bit of research and trialing in multiple VM's I have come to the conclusion that there is literally NOTHING I can do about this :cry: .  I turned off all my custom code and just ran Google Play Music inside electron and I got the exact same CPU usage as with my customization.

TLDR: Oh well :cry:

**Comment by Juburin** [comment]
Thanks for your hard work. Hopefully Google will improve Play Music. 😢

**Comment by CaffeinatedAndroid** [comment]
Experienceing high CPU load with window, after music started then paused (~5-10%).
When I enable 'Use system window borders' in Desktop Settings I get consistently much lower CPU (~0-2%).

Wasn't sure if a separate issue or not, but worth trying for anyone that stumbles on this issue.

My specs: Windows 10
